### PR TITLE
Traces pre aggregated

### PR DESCRIPTION
**Pull request by BrynCooke**
This new flag on the protobuf means that we no longer have to keep track if we are sending traces or stats. Instead we are asserting that stats are comprehensive and traces are ancillary.

From the router side we no longer skip metrics increment if a trace is being sampled.

### CODE REVIEW

File: NEXT_CHANGELOG.md [hunk 1]
```diff
@@ -41,6 +41,14 @@ It defaults to 30 seconds.
 
 By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/2271
 
+## 🐛 Fixes
+### Traces won't cause missing field-stats ([Issue #2277](https://github.com/apollographql/router/issues/2267))
+
+Previously if a request was sampled for tracing it was contributing to metrics correctly. This was a particular problem for users with a high sampling rate.
```
Comment (glasser) [inline NEXT_CHANGELOG.md:47]: "correctly"?

File: apollo-router/tests/snapshots/apollo_reports__client_name.snap [hunk 2]
```diff
@@ -589,5 +589,5 @@ traces_per_query:
     internal_traces_contributing_to_stats: []
 end_time: "[end_time]"
 operation_count: 1
-traces_pre_aggregated: false
+traces_pre_aggregated: true
```
Comment (glasser) [inline apollo-router/tests/snapshots/apollo_reports__client_name.snap:592]: A bit surprising not to see any tests where this affects the stats...

Comment (BrynCooke) [reply apollo-router/tests/snapshots/apollo_reports__client_name.snap:592]: Gonna add a new metrics integration test

Comment (BrynCooke) [reply apollo-router/tests/snapshots/apollo_reports__client_name.snap:592]: Added in [88cf26e](https://github.com/apollographql/router/pull/2277/commits/88cf26ee9fd6f70d000005ef0b505ebfd9588c1f) and [a21762b](https://github.com/apollographql/router/pull/2277/commits/a21762b37ea9272d3a54cedf646885b35afa9b26)

File: NEXT_CHANGELOG.md (hunk 1 above)
Comment (BrynCooke) [reply NEXT_CHANGELOG.md]: Fixed in [88cf26e](https://github.com/apollographql/router/pull/2277/commits/88cf26ee9fd6f70d000005ef0b505ebfd9588c1f)

File: NEXT_CHANGELOG.md [hunk 3]
```diff
@@ -41,6 +41,14 @@ It defaults to 30 seconds.
 
 By [@o0Ignition0o](https://github.com/o0Ignition0o) in https://github.com/apollographql/router/pull/2271
 
+## 🐛 Fixes
+### Traces won't cause missing field-stats ([Issue #2277](https://github.com/apollographql/router/issues/2267))
```
Comment (bnjjj) [inline NEXT_CHANGELOG.md:45]:
```suggestion
### Traces won't cause missing field-stats ([Issue #2267](https://github.com/apollographql/router/issues/2267))
```

**Review by bnjjj — APPROVED** [review]
LGTM except the typo

File: NEXT_CHANGELOG.md (hunk 3 above)
Comment (BrynCooke) [reply NEXT_CHANGELOG.md]: Fixed in [8159ad8](https://github.com/apollographql/router/pull/2277/commits/8159ad89e05bca7df75c7a1131c29a41e91cec7d)

**Comment by BrynCooke** [comment]
@bnjjj I made enough additional changes to warrant a re-review. It removes a bunch of `Option` from config to make accessing values easier, and also makes the Apollo exporter use the batch processor config.

**Review by bnjjj — APPROVED** [review]
It's cleaner ! Nice

**Pull request merged**